contracts/core/AuxiliaryBlockStore.sol

```diff
@@ -159,6 +169,95 @@ contract AuxiliaryBlockStore is BlockStore {
+     *
+     * @param _blockHash The hash of the block for which transition object
+     *                   is requested.
+     *
+     * @return coreIdentifier_ The core identifier identifies the chain that
+     *                         this block store is tracking.
+     * @return kernelHash_  The hash of the current open meta-block kernel.
+     * @return auxiliaryDynasty_ The dynasty number of auxiliary chain.
+     * @return auxiliaryBlockHash_ The block hash where the meta-block possibly
+                                   closes on the auxiliary chain.
+     * @return accumulatedGas_ The total consumed gas on auxiliary chain.
+     * @return originDynasty_ Dynasty of origin block within latest meta-block
+     *                        reported at auxiliary chain.
+     * @return originBlockHash_ Block hash of origin block within latest
+     *                          meta-block reported at auxiliary chain.
+     * @return transactionRoot_ The root of trie created by the auxiliary block
+                                store from the transaction roots of all blocks.
+     */
+    function  auxiliaryTransitionObjectAtBlock(
+        bytes32 _blockHash
+    )
+        external
+        returns(
+            bytes20 coreIdentifier_,
+            bytes32 kernelHash_,
+            uint256 auxiliaryDynasty_,
+            bytes32 auxiliaryBlockHash_,
+            uint256 accumulatedGas_,
+            uint256 originDynasty_,
+            bytes32 originBlockHash_,
+            bytes32 transactionRoot_
+        )
+    {
+        require(
+            checkpoints[_blockHash].blockHash == _blockHash,
+            "Checkpoint not defined for given block hash."
+        );
+
+        coreIdentifier_ = coreIdentifier;
+        kernelHash_ = kernelHashes[_blockHash];
+        auxiliaryDynasty_= checkpoints[_blockHash].dynasty;
+        auxiliaryBlockHash_=checkpoints[_blockHash].blockHash;
+        accumulatedGas_ = accumulatedGases[_blockHash];
+        originDynasty_ = originDynasties[_blockHash];
+        originBlockHash_ = originBlockHashes[_blockHash];
+        transactionRoot_ = accumulatedTransactionRoots[_blockHash];
+    }
+
+    /**
+     * @notice Returns auxiliary transition hash at the checkpoint defined
+     *         at given block hash.
+     *
+     * @dev It reverts transaction if checkpoint is not defined at given
+     *      block hash.
+     *
+     * @param _blockHash The hash of the block for which transition object
+     *                   is requested.
+     *
+     * @return transitionHash_ Hash of transition object at the checkpoint.
+     */
+    function  auxiliaryTransitionHashAtBlock(
+        bytes32 _blockHash
+    )
+        external
+        returns(bytes32 transitionHash_)
+    {
+        require(
+            checkpoints[_blockHash].blockHash == _blockHash,
+            "Checkpoint not defined for given block hash."
+        );
+
+        transitionHash_ = MetaBlock.hashAuxiliaryTransition(
```

Comment:
Would solidity understand `MetaBlock.hashAuxiliaryTransition(auxiliaryTransitionObjectAtBlock(_blockHash))` ? 🤔

Reply:
did you try this?

Reply:
We definetly need to change `auxiliaryTransitionObjectAtBlock` from `external` to `public`

I will check this.

Reply:
We have to keep it `external` to keep it in the interface as in the future all interface methods must be `external`. The only option would be to add a public method like `auxiliaryTransitionObjectAtBlock_()` that is called from `auxiliaryTransitionObjectAtBlock()`. Don't worry about it for now.

Reply:
Just for curiosity, I tried changing `external` to `public`. Seems solidity is not smart enough to unwrap `returns` into `arguments`. 
It gives `Wrong argument count for function call: 1 arguments given but expected 8.`